acnportal/acnsim/network/charging_network.py

```diff
@@ -163,20 +186,88 @@ def update_pilots(self, pilots, i, period):
         Station IDs not registered in the network are silently ignored.
 
         Args:
-            pilots (Dict[str, List[number]]): Dictionary mapping station_ids to lists of pilot signals. Each index in
-                the array corresponds to an a period of the simulation. [A]
+            pilots (np.Array): numpy array with a row for each station_id
+                and a column for each time. Each entry in the Array corresponds to
+                a charging rate (in A) at the staion given by the row at a time given by the column.
             i (int): Current time index of the simulation.
             period (float): Length of the charging period. [minutes]
 
         Returns:
             None
         """
-        for station_id in self._EVSEs:
-            if station_id in pilots and i < len(pilots[station_id]):
-                new_rate = pilots[station_id][i]
-            else:
-                new_rate = 0
-            self._EVSEs[station_id].set_pilot(new_rate, self._voltages[station_id], period)
+        ids = sorted(self.station_ids)
+        for station_number in range(len(ids)):
+            new_rate = pilots[station_number, i]
+            self._EVSEs[ids[station_number]].set_pilot(new_rate, self._voltages[ids[station_number]], period)
+
+    def constraint_current(self, load_currents, constraints=None, t=None, linear=False):
+        """ Return the aggregate currents subject to the given constraints. If constraints=None,
+        return all aggregate currents.
+
+        Args:
+            load_currents (Dict[str, List[number]]): Dictionary mapping load_ids to schedules of charging rates.
+            constraints (List[str]): List of constraint id's for which to calculate aggregate current. If
+                None, calculates aggregate currents for all constraints.
+            t (List[int]): List of time indices for which to calculate aggregate current. If None, 
+                calculates aggregate currents for all timesteps.
+            linear (bool): If True, linearize all constraints to a more conservative but easier to compute constraint by
+                ignoring the phase angle and taking the absolute value of all load coefficients. Default False.
+
+        Returns:
+            List[complex]: Aggregate currents subject to the given constraints.
+        """
+        # Convert list of constraint id's to list of indices in constraint matrix
+        if constraints:
+            constraint_indices = [self.constraint_index[constraint_id] for constraint_id in constraints]
+        else:
+            constraint_indices = list(self.constraint_index.values())
+        if t:
+            schedule_length = len(t)
+            schedule_matrix = np.array([[load_currents[evse_id][i] for i in t] if evse_id in load_currents else [0] * schedule_length for evse_id, _ in sorted(self._EVSEs.items())])
+        else:
+            schedule_length = len(next(iter(load_currents.values())))
```

Comment:
Make this line a bit more explicit. I think this is getting the total length of the simulation, but I'm not certain. If it is, perhaps use the Simulator property I suggested earlier.

Reply:
This line just gets the length of the schedules in `load_currents`. I changed this to
```
schedule_length = len(list(load_currents.values())[0])
```
which hopefully is clearer? I also added a comment saying what it does.